packages/shared_preferences/shared_preferences/android/src/main/java/io/flutter/plugins/sharedpreferences/MethodCallHandlerImpl.java

```diff
@@ -118,17 +123,24 @@ public void onMethodCall(MethodCall call, MethodChannel.Result result) {
 
   private void commitAsync(
       final SharedPreferences.Editor editor, final MethodChannel.Result result) {
-    new AsyncTask<Void, Void, Boolean>() {
-      @Override
-      protected Boolean doInBackground(Void... voids) {
-        return editor.commit();
-      }
+    final ExecutorService executor =
+        new ThreadPoolExecutor(0, 1, 30L, TimeUnit.SECONDS, new SynchronousQueue<Runnable>());
```

Comment:
Following up from comments on another PR: what's the reason for re-creating this every time, rather than having one instance for the object that's re-used?

Reply:
oh yes - that should be moved to the constructor.  Sorry for not catching this

Reply:
@blasten @SirusCodes Could one of you do a follow-up PR to adjust that?

Reply:
Ok I would do that👍